daffodil-test-ibm1/src/test/resources/test-suite/ibm-contributed/dpacalbin100_01.dfdl.xsd

```diff
@@ -52,7 +53,8 @@
 	<xs:element name="testBinaryMilliseconds" type="xs:dateTime">
 		<xs:annotation>
 			<xs:appinfo source="http://www.ogf.org/dfdl/">
-				<dfdl:element binaryCalendarRep="binaryMilliseconds" binaryCalendarEpoch="2009-12-24T05:15:00+05:00"/>
+				<dfdl:element binaryCalendarRep="binaryMilliseconds" binaryCalendarEpoch="2009-12-24T05:15:00+05:00"
+				  calendarPatternKind="explicit" calendarPattern="yyyy-MM-dd'T'HH:mm:ssZZZZ" />
```

Comment:
For binary calendars, patterns really only make sense for packed decimal or BCD representations. For twos-complement integer representations the DFDL spec is clear the pattern is not used.

Reply:
Gotcha. Looks like we're just passing in too many arguments into the BinaryCalendarParser then, likely because it is extending the TextCalendarParser which does need those extra arguments. I *think* we just need the calendarEv and maybe a couple others parameters. All the other stuff should go away. And it even looks like some parameters are never used, even in the TextCalendarParser. Might be good to clean some of that up while we're in this code and splitting into a CalendarBaseParser (if needed).